cmd/exporter/exporter.go

```diff
@@ -3,18 +3,37 @@ package exporter
 import (
 	"log"
 	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
 
 	"github.com/app-sre/acs-cve-prom-exporter/internal/acs"
 	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/client_golang/prometheus/promhttp"
 )
 
-func Run(endpoint string, token string) {
+func Run(endpoint string, token string) int {
 	reg := prometheus.NewRegistry()
+	exit := make(chan struct{})
+	term := make(chan os.Signal, 1)
+	signal.Notify(term, os.Interrupt, syscall.SIGTERM)
+	go acs.Run(reg, endpoint, token, exit)
 
-	go acs.Run(reg, endpoint, token)
-
-	http.Handle("/metrics", promhttp.HandlerFor(reg, promhttp.HandlerOpts{}))
-	log.Println("Beginning to serve on port :9090")
-	log.Fatal(http.ListenAndServe(":9090", nil))
+	go func(exit chan struct{}) {
+		http.Handle("/metrics", promhttp.HandlerFor(reg, promhttp.HandlerOpts{}))
+		log.Println("Beginning to serve on port :9090")
+		if err := http.ListenAndServe(":9090", nil); err != nil {
+			log.Println("Error starting server ", err)
+			exit <- struct{}{}
+		}
+	}(exit)
+	for {
+		select {
+		case <-term:
+			log.Println("shutting down gracefully")
+			return 0
+		case <-exit:
+			return 1
+		}
+	}
```

Comment:
Is this pattern the recommended way for exporter https://prometheus.io/docs/instrumenting/writing_exporters/

Reply:
@hemslo similar  pattern is followed in the other exporters in app-interface. The doc mentioned above talks very less about how to exit/terminate..  Also i would be adding  the recommendations from above doc..like config, default handler in MRs to come

Reply:
how about examples in https://prometheus.io/docs/instrumenting/writing_exporters/#maintainability-and-purity

* https://github.com/prometheus/haproxy_exporter
* https://github.com/prometheus/mysqld_exporter
* https://github.com/prometheus/node_exporter